src/transactions.rs

```diff
@@ -0,0 +1,456 @@
+            self.uids_awaiting_memos.insert(uid, txn_uid.clone());
+        }
+
+        Ok(())
+    }
+
+    /// Remove pending memo ids from the index and the stored transactions which were awaiting those ids
+    pub async fn remove_pending_uids(
+        &mut self,
+        pending_uids: Vec<u64>,
+    ) -> Result<(), KeystoreError<L>> {
+        for uid in pending_uids {
+            if let Some(txn_uid) = self.uids_awaiting_memos.remove(&uid) {
+                let txn_editor = self.get_mut(&txn_uid)?;
+                txn_editor.remove_pending_uid(uid).save()?;
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Remove a transaction from the pending index when it
+    pub async fn remove_pending(
+        &mut self,
+        timeout: u64,
+        uid: &TransactionUID<L>,
+    ) -> Result<(), KeystoreError<L>> {
+        if let Some(expiring) = self.expiring_txns.get_mut(&timeout) {
+            expiring.remove(uid);
+            if expiring.is_empty() {
+                self.expiring_txns.remove(&timeout);
+            }
+        }
+        Ok(())
+    }
+
+    /// Create an Transaction.
+    ///
+    /// When we first create a transaction it won't know all of it's values until
+    /// specific events happen on chain.  When the block for this transaction commited
+    /// We will learn it's Hash, Recept, and PendingUIDs as well as be certain of the
+    /// Fee and Asset Change records.
+    ///
+    /// Returns the editor for the created asset.
+    #[allow(clippy::too_many_arguments)]
+    pub fn create(
+        &mut self,
+        uid: TransactionUID<L>,
+        timeout: u64,
+        accounts: Vec<UserAddress>,
+        memos: Vec<Option<ReceiverMemo>>,
+        sig: Signature,
+        freeze_outputs: Vec<RecordOpening>,
+        inputs: Vec<RecordOpening>,
+        outputs: Vec<RecordOpening>,
+        time: DateTime<Local>,
+        asset: AssetCode,
+        kind: TransactionKind<L>,
+        senders: Vec<UserAddress>,
+        receivers: Vec<(UserAddress, RecordAmount)>,
+    ) -> Result<TransactionEditor<'_, L>, KeystoreError<L>> {
+        let txn = Transaction::<L> {
+            uid,
+            timeout,
+            hash: None,
+            pending_uids: HashSet::new(), //pending_uids
+            accounts,
+            memos,
+            sig,
+            /// If the transaction is a freeze, the expected frozen/unfrozen outputs.
+            freeze_outputs,
+            inputs,
+            outputs,
+            time,
+            asset,
+            kind,
+            senders,
+            receivers,
+            fee_change: None,   // fee_change
+            asset_change: None, // asset_change
+            receipt: None,      // receipt
+        };
+        self.expiring_txns
+            .entry(timeout)
+            .or_insert_with(HashSet::default)
+            .insert(txn.uid().clone());
+        let mut editor = TransactionEditor::new(&mut self.store, txn);
+        editor.save()?;
```

Comment:
I think maybe we can just return an error here if the UID already exists? For assets we have the updating thing because the same asset can be loaded from multiple sources (e.g. manually imported by the user and also loaded from an official asset library) but I don't believe we ever need to call `create` on a transaction that already exists?

Reply:
That's right we should not call create on an asset that exists in any situation.  All the updates should be done via the editors